src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubScaleMonitor.cs

```diff
@@ -0,0 +1,395 @@
+                return metrics;
+            }
+
+            // Get the PartitionRuntimeInformation for all partitions
+            var tasks = new Task<EventHubPartitionRuntimeInformation>[runtimeInfo.PartitionCount];
+
+            for (int i = 0; i < runtimeInfo.PartitionCount; i++)
+            {
+                tasks[i] = _client.GetPartitionRuntimeInformationAsync(i.ToString());
+            }
+
+            await Task.WhenAll(tasks);
+
+            return await CreateTriggerMetrics(tasks.Select(t => t.Result).ToList());
+        }
+
+        internal async Task<EventHubTriggerMetrics> CreateTriggerMetrics(List<EventHubPartitionRuntimeInformation> partitionRuntimeInfo, bool alwaysLog = false)
+        {
+            long totalUnprocessedEventCount = 0;
+            bool logPartitionInfo = alwaysLog ? true : DateTime.UtcNow >= _nextPartitionLogTime;
+            bool logPartitionWarning = alwaysLog ? true : DateTime.UtcNow >= _nextPartitionWarningTime;
+
+            // For each partition, get the last enqueued sequence number.
+            // If the last enqueued sequence number does not equal the SequenceNumber from the lease info in storage,
+            // accumulate new event counts across partitions to derive total new event counts.
+            List<string> partitionErrors = new List<string>();
+            for (int i = 0; i < partitionRuntimeInfo.Count; i++)
+            {
+                long partitionUnprocessedEventCount = 0;
+
+                Tuple<BlobPartitionLease, string> partitionLeaseFile = await GetPartitionLeaseFileAsync(i);
+                BlobPartitionLease partitionLeaseInfo = partitionLeaseFile.Item1;
+                string error = partitionLeaseFile.Item2;
+
+                if (partitionRuntimeInfo[i] == null || partitionLeaseInfo == null)
+                {
+                    partitionErrors.Add(error);
+                }
+                else
+                {
+                    // Check for the unprocessed messages when there are messages on the event hub parition
+                    // In that case, LastEnqueuedSequenceNumber will be >= 0
+                    if ((partitionRuntimeInfo[i].LastEnqueuedSequenceNumber != -1 && partitionRuntimeInfo[i].LastEnqueuedSequenceNumber != partitionLeaseInfo.SequenceNumber)
+                        || (partitionLeaseInfo.Offset == null && partitionRuntimeInfo[i].LastEnqueuedSequenceNumber >= 0))
+                    {
+                        partitionUnprocessedEventCount = GetUnprocessedEventCount(partitionRuntimeInfo[i], partitionLeaseInfo);
+                        totalUnprocessedEventCount += partitionUnprocessedEventCount;
+                    }
+                }
+
+                if (logPartitionInfo)
+                {
+                    _logger.LogInformation($"Function '{_functionId}', Partition: {i}, Total new partition events: {partitionUnprocessedEventCount}");

```

Comment:
The host will log the metrics you return already [here](https://github.com/Azure/azure-functions-host/blob/398f8264f2d579a71d31ba3f59a442afbbdef643/src/WebJobs.Script/Scale/FunctionsScaleMonitorService.cs#L100). If this is a duplicate log, perhaps we don't need this one? Check the logs on one of your runs to see if this is the case.